pkg/rehearse/metrics.go

```diff
@@ -0,0 +1,101 @@
+package rehearse
+
+import (
+	"encoding/json"
+	"io/ioutil"
+
+	"github.com/sirupsen/logrus"
+	prowconfig "k8s.io/test-infra/prow/config"
+
+	"github.com/openshift/ci-operator-prowgen/pkg/config"
+)
+
+type ExecutionMetrics struct {
+	SubmittedRehearsals []string `json:"submitted"`
+	FailedRehearsals    []string `json:"failed"`
+	PassedRehearsals    []string `json:"successful"`
+}
+
+type Metrics struct {
+	Org  string `json:"org"`
+	Repo string `json:"repo"`
+	Pr   int    `json:"pr"`
+
+	ChangedCiopConfigs []string `json:"changed_ciop_configs"`
+	ChangedPresubmits  []string `json:"changed_presubmits"`
+	ChangedTemplates   []string `json:"changed_templates"`
+
+	// map a job name to a list of reasons why we want to rehearse it
+	Opportunities map[string][]string `json:"opportunities"`
+	Actual        []string            `json:"actual"`
+
+	Execution *ExecutionMetrics `json:"execution"`
+
+	logger logrus.Entry
+	file   string
+}
+
+func NewMetrics(file string) *Metrics {
+	return &Metrics{
+		ChangedCiopConfigs: []string{},
+		ChangedPresubmits:  []string{},
+		ChangedTemplates:   []string{},
+
+		Opportunities: map[string][]string{},
```

Comment:
lets `make()` this

Reply:
Why? It seems that most of our code is using literals...